### PR TITLE
Internal: Create step - Set up a header [ED-15445]

### PR DESCRIPTION
**Pull request by ronenboxer**

### CODE REVIEW

File: modules/checklist/assets/js/components/checklist-card-content.js [hunk 1]
```diff
@@ -17,8 +33,16 @@ const ChecklistCardContent = ( props ) => {
 				</Typography>
 			</CardContent>
 			<CardActions>
-				<Button size="small" color="secondary" variant="text" className="mark-as-done">{ __( 'Mark as done', 'elementor' ) }</Button>
-				<Button size="small" variant="contained" className="cta-button">{ CTA }</Button>
+
```
Comment (elchugreeva) [inline modules/checklist/assets/js/components/checklist-card-content.js:36]: Why do we remove buttons from component?  Maybe we should render mark as done conditionally, based on isLocked?

Comment (ronenelementor) [reply modules/checklist/assets/js/components/checklist-card-content.js:36]: It wasn't removed, just arranged differently for readability

File: modules/checklist/steps-manager.php [hunk 2]
```diff
@@ -159,6 +163,6 @@ private function get_step_instance( string $step_id ) : ?Step_Base {
 	 * @return array
 	 */
 	private static function get_step_ids() : array {
-		return [ Create_Pages::STEP_ID ];
+		return [ Create_Pages::STEP_ID, Setup_Header::STEP_ID ];
```
Comment (louiswol94) [inline modules/checklist/steps-manager.php:166]: I am not sure how this method is supposed to work. How does 3rd party steps get in here?

File: modules/checklist/steps/setup-header.php [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+class Setup_Header extends Step_Base {
+	const STEP_ID = 'setup_header';
+
+	public function __construct( ...$args ) {
+		call_user_func( [ parent::class, __FUNCTION__ ], ...$args );
```
Comment (louiswol94) [inline modules/checklist/steps/setup-header.php:16]: Why not just `parent::__construct` ?

File: modules/checklist/steps/setup-header.php [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+					'compare' => '=',
+				],
+				[
+					'key' => '_elementor_conditions',
+				],
+			],
+			'posts_per_page' => 1,
+		];
+		$query = $this->wordpress_adapter->get_query( $args );
+		$headers = $query->posts ?? [];
```
Comment (louiswol94) [inline modules/checklist/steps/setup-header.php:42]:
```suggestion
		$header_templates = $query->posts ?? [];
```

File: modules/checklist/steps/setup-header.php [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+					'key' => '_elementor_template_type',
+					'value' => 'header',
+					'compare' => '=',
+				],
+				[
+					'key' => '_elementor_conditions',
+				],
+			],
+			'posts_per_page' => 1,
+		];
```
Comment (louiswol94) [inline modules/checklist/steps/setup-header.php:40]: Please add the following args:

```
'fields' => 'ids',
'no_found_rows' => true,
'update_post_term_cache' => false,
'update_post_meta_cache' => false,
```
(for performance)

File: modules/checklist/steps/step-base.php [hunk 6]
```diff
@@ -11,6 +11,13 @@
 }
 
 abstract class Step_Base {
+	const TIER_CORE = 'core';
+	const TIER_ESSENTIAL = 'essential';
+	const TIER_ADVANCED = 'advanced';
+	const TIER_EXPERT = 'expert';
+	const TIER_AGENCY = 'agency';
```
Comment (louiswol94) [inline modules/checklist/steps/step-base.php:18]: Do we maybe have these already another place in the code base?

**Review by louiswol94 — CHANGES_REQUESTED** [review]

File: modules/checklist/steps/step-base.php (hunk 6 above)
Comment (ronenelementor) [reply modules/checklist/steps/step-base.php:18]: `modules/connect/module`
Is this global enough ti use?

File: modules/checklist/steps/setup-header.php (hunk 5 above)
Comment (ronenelementor) [reply modules/checklist/steps/setup-header.php:40]: Thanks!

File: modules/checklist/assets/js/components/checklist-card-content.js [hunk 7]
```diff
@@ -1,8 +1,24 @@
...
+		description,
+		learn_more_url: learnMoreUrl,
+		learn_more_text: learnMoreText,
+		image_src: imageSrc,
+		is_locked: isLocked,
+		promotion_url: promotionUrl,
+		cta_url: ctaUrl,
+	} = step.config;
+
+	const ctaText = isLocked ? __( 'Upgrade Now', 'elementor-pro' ) : step.config.cta_text,
```
Comment (louiswol94) [inline modules/checklist/assets/js/components/checklist-card-content.js:16]: Do you need to `import { __ } from '@wordpress/i18n';`

File: modules/checklist/steps/setup-header.php (hunk 4 above)
Comment (ronenelementor) [reply modules/checklist/steps/setup-header.php:42]: Done

File: modules/checklist/steps/setup-header.php (hunk 3 above)
Comment (ronenelementor) [reply modules/checklist/steps/setup-header.php:16]: Done!

File: modules/checklist/steps-manager.php (hunk 2 above)
Comment (ronenelementor) [reply modules/checklist/steps-manager.php:166]: I'll make a static property, and when the filter is implemented we'll merge or do some kind of operation, wdyt?

File: modules/checklist/assets/js/components/checklist-card-content.js (hunk 7 above)
Comment (ronenelementor) [reply modules/checklist/assets/js/components/checklist-card-content.js:16]: I thought about that, it didn't offer my the import and it seemed to work, do you want me to import it just to be on the safe side?

File: modules/checklist/steps/step-base.php (hunk 6 above)
Comment (louiswol94) [reply modules/checklist/steps/step-base.php:18]: I would maybe extract to some common area and re-use it.

File: modules/checklist/assets/js/components/checklist-card-content.js (hunk 7 above)
Comment (louiswol94) [reply modules/checklist/assets/js/components/checklist-card-content.js:16]: Import it rather

File: modules/checklist/steps-manager.php (hunk 2 above)
Comment (ronenelementor) [reply modules/checklist/steps-manager.php]: Independent task for that

File: modules/checklist/assets/js/components/checklist-card-content.js (hunk 1 above)
Comment (elchugreeva) [reply modules/checklist/assets/js/components/checklist-card-content.js:36]: @ronenelementor I see 1 button in cards actions, instead of 2

Comment (ronenelementor) [reply modules/checklist/assets/js/components/checklist-card-content.js:36]: NICE CATCH!

File: modules/checklist/steps/step-base.php (hunk 6 above)
Comment (ronenelementor) [reply modules/checklist/steps/step-base.php]: I created a `constants` file in the `core/utils` folder

File: core/utils/constants.php [hunk 8]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace Elementor\Core\Utils;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+class Constants {
```
Comment (louiswol94) [inline core/utils/constants.php:9]:
```suggestion
abstract class Constants {
```
We don't want any instances of this class :)

Comment (ronenelementor) [reply core/utils/constants.php:9]: Perfect

**Review by louiswol94 — APPROVED** [review]

**Review by elchugreeva — APPROVED** [review]

**Pull request merged**